forest/sycamore/responses.py

```diff
@@ -171,3 +182,117 @@ def get_most_common_answer(col):
 
     detail = detail[detail_cols]
     return detail, out
+
+
+def format_responses_by_submission(input_agg_data: pd.DataFrame) -> dict:
+    """Format survey answers with one line per survey submission
+
+    For each survey, create a DataFrame with one column for each question, and
+    one row for each submission.
+
+    Args:
+        input_agg_data(DataFrame): output from aggregate_surveys_config or
+            aggregate_surveys_no_config
+
+    Returns:
+        Dict with a key for each survey ID in agg_data. Each value is a
+        dataframe with readable survey submission information.
+
+    """
+    agg_data = input_agg_data.copy()
```

Comment:
Is copying necessary here? I don't see where the dataframe is getting modified.

Reply:
Yes, you're correct. My thinking had been that it was good to make a copy of all input data frames in case a function modified one, but that does unnecessarily use resources.